src/probe_modules/module_tcp_synscan.c

```diff
@@ -107,14 +107,22 @@ int synscan_validate_packet(const struct ip *ip_hdr, uint32_t len,
 	}
 	// validate tcp acknowledgement number
 	if (htonl(tcp->th_ack) != htonl(validation[0])+1) {
-		return 0;
+
+		// RSTs do not need to +1 the ack
```

Comment:
I think the way this logic is arranged is currently confusing. It would be clearer if we had something like
if RST: apply these rules
else: apply these other rules.

Right now, this is effectively,
apply rules, if this one random rule fails, but you're a RST, just ignore that branch of the logic. While these are arguably equivalent, I believe that the former is cleaner than the latter.